application/src/main/java/org/thingsboard/server/service/ttl/KafkaEdgeTopicsCleanUpService.java

```diff
@@ -0,0 +1,111 @@
+
+    private static final long ONE_MONTH_MILLIS = TimeUnit.DAYS.toChronoUnit().getDuration().multipliedBy(30).toMillis();
+
+    private final EdgeService edgeService;
+    private final TenantService tenantService;
+    private final AttributesService attributesService;
+
+    private final TopicService topicService;
+    private final PartitionService partitionService;
+
+    private final TbKafkaSettings kafkaSettings;
+    private final TbKafkaTopicConfigs kafkaTopicConfigs;
+
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor(ThingsBoardThreadFactory.forName("kafka-edge-topic-cleanup"));
+
+    @Scheduled(initialDelayString = "#{T(org.apache.commons.lang3.RandomUtils).nextLong(0, ${sql.ttl.edge_events.execution_interval_ms})}", fixedDelayString = "${sql.ttl.edge_events.execution_interval_ms}")
+    public void cleanUp() {
+        executorService.submit(() -> {
+            PageDataIterable<TenantId> tenants = new PageDataIterable<>(tenantService::findTenantsIds, 10_000);
+            for (TenantId tenantId : tenants) {
+                try {
+                    cleanUp(tenantId);
+                } catch (Exception e) {
+                    log.warn("Failed to drop kafka topics for tenant {}", tenantId, e);
+                }
+            }
+        });
+    }
+
+    private void cleanUp(TenantId tenantId) throws Exception {
+        if (!partitionService.resolve(ServiceType.TB_CORE, tenantId, tenantId).isMyPartition()) {
+            return;
+        }
+
+        PageDataIterable<EdgeId> edgeIds = new PageDataIterable<>(link -> edgeService.findEdgeIdsByTenantId(tenantId, link), 1024);
+        long currentTimeMillis = System.currentTimeMillis();
+
+        for (EdgeId edgeId : edgeIds) {
+            Optional<AttributeKvEntry> attributeOpt = attributesService.find(tenantId, edgeId, AttributeScope.SERVER_SCOPE, DefaultDeviceStateService.LAST_CONNECT_TIME).get();
+            if (attributeOpt.isPresent()) {
+                Optional<Long> lastConnectTimeOpt = attributeOpt.get().getLongValue();
+                if (lastConnectTimeOpt.isPresent() && isTopicExpired(lastConnectTimeOpt.get(), currentTimeMillis)) {
+                    String topic = topicService.buildEdgeEventNotificationsTopicPartitionInfo(tenantId, edgeId).getTopic();
+                    TbKafkaAdmin kafkaAdmin = new TbKafkaAdmin(kafkaSettings, kafkaTopicConfigs.getEdgeEventConfigs());
+                    if (kafkaAdmin.isTopicEmpty(topic)) {
+                        kafkaAdmin.deleteTopic(topic);
+                        log.info("Removed outdated topic for tenant {} and edge with id {} older than {}", tenantId, edgeId, Date.from(Instant.ofEpochMilli(currentTimeMillis - ONE_MONTH_MILLIS)));
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean isTopicExpired(long lastConnectTime, long currentTimeMillis) {
+        return lastConnectTime + ONE_MONTH_MILLIS < currentTimeMillis;
```

Comment:
I think we should use `sql.ttl.edge_events.edge_events_ttl` or some other **configurable** value